Flaky E2E: refactor Domain Upsell from Home dashboard.

worldomonation opened the pull request:
Fixes https://github.com/Automattic/wp-calypso/issues/77007.

## Proposed Changes

This PR refactors the `Domain Upsell: My Home` spec to be more stable.

Key changes:
- add wait before interacting with the upsell button, which is likely the source of https://github.com/Automattic/wp-calypso/issues/77007.
- remove single-purpose methods in favor of general purpose methods.

## Testing Instructions

Changes survive a 100 iteration loop:
<img width="1029" alt="image" src="https://github.com/Automattic/wp-calypso/assets/6549265/4e83e270-df29-4640-89d8-1d08934f19dd">

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

worldomonation committed:
test/e2e/specs/domain-upsell/domain-upsell__my-home.ts
- rename the suite.
- remove steps where appropriate.
- update spec to use role-based selectors.

worldomonation committed:
specs/domain-upsell/domain-upsell__my-home.ts
- better handling of the asynchronous load of the home dashboard.

youbrokesomething reviewed (approved):
This looks good to me. 🚢 
@worldomonation is there anything here that you'd suggest to use as an example for training or feedback?

worldomonation commented:
> @worldomonation is there anything here that you'd suggest to use as an example for training or feedback?

A few things to note:
- I'd like to have a discussion that we agree this is the way to go for future E2E scripts.
- our documents need to be updated to reflect this shift in paradigm.

I think a quick P2 post is in order. I will draft one up this week.

worldomonation committed:
test/e2e/specs/domain-upsell/domain-upsell__my-home.ts
- use POMs

packages/calypso-e2e/src/lib/pages/my-home-page.ts
- add new generic methods.

worldomonation committed:
packages/calypso-e2e/src/lib/pages/plans-page.ts
- remove dead code

worldomonation commented:
@youbrokesomething Heads up that I've revised the PR, removing much of the "big change" that I was alluding to. 

I feel that it is preferable to start discussion in a P2 first, before moving to remove POMs.
